src/views/Home.jsx

```diff
@@ -1,14 +1,57 @@
 import './Home.css';
+import { useState, useCallback } from 'react';
+import { getItemData, streamListItems } from '../api';
+import { generateToken } from '@the-collab-lab/shopping-list-utils';
+
+export function Home({ setListToken }) {
+	const [tokenInput, setTokenInput] = useState('');
+
+	const handleClick = useCallback(() => {
+		const newToken = generateToken();
+		setListToken(newToken);
+	}, [setListToken]);
+
+	const tokenHandler = (e) => {
+		setTokenInput(e.target.value);
+	};
+
+	const handleTokenSubmit = (e) => {
+		e.preventDefault();
+		streamListItems(tokenInput, (snapshot) => {
+			const listData = getItemData(snapshot);
+			if (listData?.length === 0) {
+				alert('this token does not exist');
+			} else {
+				setListToken(tokenInput);
+			}
+		});
+	};
 
-export function Home({ onClick }) {
 	return (
 		<div className="Home">
 			<p>
 				Hello from the home (<code>/</code>) page!
 			</p>
-			<button onClick={onClick} className="createListBtn">
+			<button onClick={handleClick} className="createListBtn">
 				Create a new list
 			</button>
+
+			<form onSubmit={handleTokenSubmit}>
+				<label htmlFor="join-list">
+					Enter a three-words token to join existing shopping list
+				</label>
+				<input
+					type="text"
+					onChange={tokenHandler}
+					value={tokenInput}
+					name="join-list"
+					id="join-list"
+				/>
+
+				<button type="submit" disabled={!tokenInput}>
```

Comment:
I LOVE that you have the button disabled until the user adds a token. So neat! ⭐️

This might be over kill, but maybe we could utilize regex to keep the button disabled unless the token matches the lower case three-hyphenseparated-word format?

I think the regex for that would be:
`^([a-z]+)-([a-z]+)-([a-z]+)$`

Reply:
Cool idea!

Reply:
Wow, great idea, @itsoliviasparks! It looks like the token generation is creating 3 words without spaces currently, but regex could be used to check for 3 words!

A related thing that could be done to check user validation is to use the `trim()` method on the input to remove leading and trailing white space. Right now, the list `lawn annal sent` is valid, but if someone accidentally added a space before or after, it would come back as invalid.